components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSOAPUtils.java

```diff
@@ -0,0 +1,281 @@
+                samlRequest = Base64.getEncoder().encodeToString(message.getBytes());
+                elementSize += 1;
+            }
+        } catch (NullPointerException e) {
+            String err = "SOAP message body cannot be null";
+            log.error(err);
+            throw new IdentitySAML2ECPException(e.getMessage());
+        }
+        if (elementSize == 0) {
+            String err = "The SOAP message body is Empty";
+            log.error(err);
+            throw new IdentitySAML2ECPException(err);
+        } else if (elementSize == 1) {
+            try {
+                unmarshall(message);
+            } catch (IdentityException e) {
+                String err =  "SOAP Message doesn't contain a valid SAML Request";
+                log.warn(err);
+                throw new IdentitySAML2ECPException(err);
+            }
+        } else if (elementSize > 1) {
+            String err = "SOAP Message contains more than one XML Element";
+            log.error(err);
+            throw new IdentitySAML2ECPException(err);
+        }
+        return samlRequest;
+    }
+
+    /**
+     *
+     * Creates a SOAP Fault message including the fault coe and fault string.
+     * @param faultString detailed error message
+     * @param faultcode
+     * @return
+     */
+    public static String createSOAPFault(String faultString, String faultcode) {
+        SOAPMessage soapMsg =  null;
+        try {
+            MessageFactory factory = MessageFactory.newInstance();
+            soapMsg = factory.createMessage();
+            SOAPPart part = soapMsg.getSOAPPart();
+            SOAPEnvelope envelope = part.getEnvelope();
+            SOAPBody body = envelope.getBody();
+            SOAPFault fault = body.addFault();
+            fault.setFaultString(faultString);
+            fault.setFaultCode(new QName("http://schemas.xmlsoap.org/soap/envelope/", faultcode));
+
+        } catch (SOAPException e) {
+            String err = "SOAP Exception when creating SOAP fault";
+            log.error(err);
+        }
+        if (soapMsg != null) {
+            return convertSOAPToString(soapMsg).replace("<?xml version=\"1.0\" encoding=\"UTF-8\"?>", "");
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     *
+     * @param samlRes
+     * @param acUrl
+     * @return
+     */
+    public static String createSOAPMessage(String samlRes, String acUrl) {
+        SOAPMessage soapMsg = null;
+        try {
+            MessageFactory factory = MessageFactory.newInstance();
+            soapMsg = factory.createMessage();
+            SOAPPart part = soapMsg.getSOAPPart();
+            SOAPEnvelope envelope = part.getEnvelope();
+            SOAPHeader header = envelope.getHeader();
+            SOAPHeaderElement soapHeaderElement = header.addHeaderElement(envelope.createName("Response",
+                    "ecp",
+                    "urn:oasis:names:tc:SAML:2.0:profiles:SSO:ecp"));
```

Comment:
Let's define all these constants as constants.

Reply:
defined the constants